### ISSUE TITLE
Missing type annotations for `wallet-js` index.js file

### ISSUE THREAD

**Issue by refi93** [input-output-hk/catalyst-core#131]
I tried to build latest `wallet-js` (commit 340401bb5b1e5bdcf7b01a86cdecf11729ff15bd) with the following command:

```
wasm-pack build -d pkg --target nodejs --out-name wallet-js
```

However, when I check the built ts package d.ts files I can't find type annotations (`*d.ts` files) for the functions exposed in index.js:

 https://github.com/input-output-hk/catalyst-core/blob/main/src/chain-wallet-libs/bindings/wallet-js/js/index.js
 
Seems like the index.js is missing proper type annotations. Or am I perhaps missing something?

**Comment by refi93** [comment]
maybe it's just not clear to me how to include the `index.js` file in the final package to be used: https://github.com/input-output-hk/catalyst-core/commit/5bf14f2d411911a1d564a16c4147313f3a6f27dc

**Comment by refi93** [comment]
additional question: Could you clarify why the `initValue` is relevant to construct the wallet object? I guess it's supposed to be the balance of the wallet, but I don't see it playing any role in signing of the transaction

https://github.com/input-output-hk/catalyst-core/blob/340401bb5b1e5bdcf7b01a86cdecf11729ff15bd/src/chain-wallet-libs/bindings/wallet-js/js/index.js#L52

**Comment by minikin** [comment]
@Mr-Leshiy we need your help here.

**Comment by Mr-Leshiy** [comment]
>additional question: Could you clarify why the initValue is relevant to construct the wallet object? I guess it's supposed to be the balance of the wallet, but I don't see it playing any role in signing of the transaction.

yes, you are correct, it is wallet balance, it is needed to correctly track and establish transaction fees which should be set during the generation process.

>Also, when signing real transactions, shouldn't the spending counter(s) be in sync with the state of the wallet on the network, and hence somehow passed to the wallet? Currently, they seem to be hardcoded to 1 in the logic constructing the wallet

Good point, need to take a look how to update or fix it.


>However, when I check the built ts package d.ts files I can't find type annotations (*d.ts files) for the functions exposed in index.js: https://github.com/input-output-hk/catalyst-core/blob/main/src/chain-wallet-libs/bindings/wallet-js/js/index.js

Yes, you are correct I have not added typings for this package. Code from that package actually a wrapper over Wasm wallet package.

**Comment by refi93** [comment]
> Good point, need to take a look how to update or fix it.

btw, I see that the wallet [tries to set up 8 lanes](https://github.com/input-output-hk/catalyst-core/blob/340401bb5b1e5bdcf7b01a86cdecf11729ff15bd/src/chain-wallet-libs/bindings/wallet-js/js/index.js#L56), each one with its own spending counter, and if I try to set up just one "lane", I get an `invalid spending counters provided to the set account state function` error - why are multiple lanes needed? Isn't the "basic" use case to have just one lane to sign the votes sequentially?

**Comment by refi93** [comment]
> > Good point, need to take a look how to update or fix it.
> 
> btw, I see that the wallet [tries to set up 8 lanes](https://github.com/input-output-hk/catalyst-core/blob/340401bb5b1e5bdcf7b01a86cdecf11729ff15bd/src/chain-wallet-libs/bindings/wallet-js/js/index.js#L56), each one with its own spending counter, and if I try to set up just one "lane", I get an `invalid spending counters provided to the set account state function` error - why are multiple lanes needed? Isn't the "basic" use case to have just one lane to sign the votes sequentially?

made a follow up issue for this: https://github.com/input-output-hk/catalyst-core/issues/139
